project/BuildSettings.scala

```diff
@@ -64,6 +63,20 @@ object BuildSettings {
     }.taskValue
   )
 
+  def scalacOptionsVersion(scalaVersion: String): Seq[String] = {
+    // We need to use '-Ywarn-macros:default' flag due to following issue in specs2
+    // https://github.com/etorreborre/specs2/issues/1237
+    // However, Scala 2.12.18 doesn't allow to set it to 'default' therefore we need to
+    // continue to use 'after' with it.
+    val versionSpecificOptions = CrossVersion.partialVersion(scalaVersion) match {
+      case Some((2, scalaMajor)) if scalaMajor == 12 => "-Ywarn-macros:after"
+      case _                                         => "-Ywarn-macros:default"
+    }
+    Seq(
+      "-Wconf:origin=scala.collection.compat.*:s"
+    ) :+ versionSpecificOptions
+  }
+
```

Comment:
See [here](https://github.com/snowplow/stream-collector-private/commit/cfdc6b42e4cb2704414fd9e1a0cc63cc3d50e0e1#diff-a7069678b490aad347fac8a256daeda5f83135c60bbec6239eebc507873baaca) where we already did something similar in another repo.  For 2.13 you could omit the `-Ywarn-macros:default` because it is the default anyway.

...and once you do that, then it kinda changes what the code comment needs to say.  Because you no longer need to explain why 2.13 is set to something special (it's not special!).  You just need to explain why 2.12 is set to something non-standard.